src/pudl/metadata/classes.py

```diff
@@ -1872,14 +1872,24 @@ def get_sorted_resources(self) -> StrictList(Resource):
         on their name which results in the following order to promote output
         tables to users and push intermediate tables to the bottom of the
         docs: output, core, intermediate.
-
         In the future we might want to have more fine grain control over how
         Resources are sorted.
 
         Returns:
             A sorted list of resources.
         """
-        return sorted(self.resources, key=lambda r: r.name, reverse=True)
+        resources = self.resources
+
+        def sort_resource_names(resource: Resource):
+            pattern = re.compile(r"(_out_|out_|core_)")
```

Comment:
I think this regex should specify that these are the beginnings of the table names, otherwise we might get some inadvertent matches with tables like `core_eia__flout_the_authority` or out_ferc1__hardcore_energy`
```suggestion
            pattern = re.compile(r"^(_out_|out_|core_)")
```